packages/agents/src/AgentBase.ts

```diff
@@ -96,6 +96,12 @@ export class AgentBase<TRunArgs, TAgentBaseContext extends AgentBaseContext> imp
         chat.persistent(message);
       });
 
+      // Add an extra prompt informing agent about variable usage
+      chat.persistent({
```

Comment:
Agreed the most recent persistent message should be the goal.

I found that without this, the LLM wasn't able to use variables correctly. Two problems I saw:
1. It wouldn't know that it could pass variables names into functions as arguments.
2. It would pass "variable-name" into functions and not "${variable-name}"